youtube_dl/extractor/bbc.py

```diff
@@ -959,6 +959,16 @@ def _real_extract(self, url):
              r'videoId\s*:\s*["\'](%s)["\']' % self._ID_REGEX],
             webpage, 'vpid', default=None)
 
+        # bbc reel (e.g. https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness)
+        initial_data = self._search_regex(
+            r'<script[^>]+id="initial-data"[^>]+data-json=\'(.+)\'>',
+            webpage, 'initial data', fatal=False, default=None)
+        if initial_data:
+            programme_id = self._search_regex(
+                r'"versionID":"(%s)"' % self._ID_REGEX,
+                unescapeHTML(initial_data),
+                'programme id', fatal=False, default=None)
```

Comment:
parse json data and extract `versionID` from there.

Reply:
The json is deeply nested. Something like JSONPath would have helped. I used an auxiliary function to parse it, but really it would have been easier just to pull the pid from the URI: https://github.com/dirkf/youtube-dl-1/commit/cccbb670f0fe1546c6e62f8da7fe545dc50bb9f1